clientdb/order.go

```diff
@@ -34,6 +34,10 @@ const (
 	// notAllowedNodeIDsType is the tlv type we use to store the list of
 	// node ids the order is not allowed to match with.
 	notAllowedNodeIDsType tlv.Type = 5
+
+	// privateChannelType is the tlv type we use to store the flag for
+	// private channel order constrains.
+	privateChannelType tlv.Type = 6
```

Comment:
I think one other option here is to: expose this as either a channel type, or channel type modifier. In the former case, it'd be an entirely new time. In the latter case, it would be part of another (?) bitvector that specified modifications on the base channel type.